### PR TITLE
feature: Offer undo button for recently-deleted snooze entries

### LINKED ISSUE #72: Add confirmation when deleting snoozed tab?
I just had the issue where I accidentally+eagerly double clicked on a delete icon when deleting a snoozed tab, and accidentally deleted the subsequent snoozed tab which moved to fill the first snoozed tab's position in the list.

Not sure if we want to add some confirmation step, or if that makes the deleting step too interactive.

### PR DESCRIPTION
**Pull request by lmorchard**
- Delete button deletes item, but stores it in a temporary structure for
  restore.

- Temporary items are displayed faded out with an undo button that
  re-schedules it from the temporary structure.

- CSS tweaks to fade out deleted item

- Clear recently deleted items after 5 seconds

- Move idForItem to lib/utils as a shared function

- Add src/icons/undo.svg from Noun Project 716798

- Include credit for new icon in README.md

Fixes #72.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,8 +54,10 @@ Many of our icons came from [the Noun Project][nouns]:
 * Weekend: "[Sunglasses][sunglasses]" By NAS, from the Noun Project.
 * Next Month: "[Calendar][calendar]" By artworkbean, from the Noun Project. (Modified)
 * Pick a Date: "[Calendar][calendar]" By artworkbean, from the Noun Project.
+* Undo: "[Undo][]" By H Alberto Gongora, from the Noun Project.
 
 
+[undo]: https://thenounproject.com/search/?q=undo&i=716798
```
Comment (bwinton) [inline README.md]: We should leave these in alphabetical order, since I went to all that effort in the first place…  😉

Comment (lmorchard) [inline README.md]: They're not really in alpha order now, but I can sort them!

**Comment by lmorchard** [comment]
Yeah, tried to figure out how to transition items entering & leaving the list, but that'll be more involved. I think there's something that could be done with `ReactCSSTransitionGroup`. Kind of want to push that off to another PR, but can keep banging on it for this one

**Comment by lmorchard** [comment]
Just tried adding a transition - let's see if that works better. Still seemed to jump every now and then on my machine, but seemed better

**Comment by bwinton** [comment]
With the new proptypes landed, this now fails linting with the error:
> …\src\lib\components\ManagePanel.js
>  199:13  error  'undeleteSnoozedTab' is missing in props validation  react/prop-types
🙂

And we might as well alphabetize those while we're at it.  (I thought I did in my original patch, but seem to have forgotten to.)

**Comment by lmorchard** [comment]
Updated to sort the things, fix the linting, and will merge assuming tests are green

**Pull request merged**